debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlStreamingChangeEventSource.java

```diff
@@ -229,16 +229,22 @@ protected void onEvent(MySqlOffsetContext offsetContext, Event event) {
     private void setEventTimestamp(Event event, long eventTs) {
         // Prefer higher resolution replication timestamps from MySQL 8 GTID events, if possible
         if (isGtidModeEnabled) {
-            if (event.getHeader().getEventType() == EventType.GTID) {
-                GtidEventData gtidEvent = unwrapData(event);
-                final long gtidEventTs = gtidEvent.getOriginalCommitTimestamp();
-                if (gtidEventTs != 0) {
-                    // >= MySQL 8.0.1, prefer the higher resolution replication timestamp
-                    eventTimestamp = Instant.EPOCH.plus(gtidEventTs, ChronoUnit.MICROS);
-                }
-                else {
-                    // Fallback to second resolution event timestamps
-                    eventTimestamp = Instant.ofEpochMilli(eventTs);
+            if (connection.isMariaDb()) {
+                // Fallback to second resolution event timestamps
+                eventTimestamp = Instant.ofEpochMilli(eventTs);
+            }
```

Comment:
I'm not extremely happy with using this here; however, since the adapter does not provide access to the unwrap method for the event data, it's probably the cleanest solution for now and keeps the change localized, so I think it's fine until I refactor this in 2.7.  :+1:

Reply:
Agree. Also possible to use the `MARIADB_GTID` event type. I looked at metadata exposed by [MariaDB GTID events](https://mariadb.com/kb/en/gtid_event/) and its quite sparse, unfortunately.

```java
if (event.getHeader().getEventType() == EventType.MARIADB_GTID) {
    // Fallback to second resolution event timestamps
    eventTimestamp = Instant.ofEpochMilli(eventTs);
}
```

Reply:
💯 

> That said, I wonder if this method would be better written this way, it seems more concise in terms of logic and self documenting in my opinion, wdyt:

How do you prefer to move forward? Apply direct or want me to commit the suggestion?

Reply:
Please go ahead and if you have the cycles @methodmissing

Reply:
Applied in https://github.com/debezium/debezium/pull/5281/commits/853f39b43c1e78c022444036199c8ffca9e66d4c and piped though a lower environment - no regression:

<img width="1122" alt="Screenshot 2024-02-17 at 15 05 37" src="https://github.com/debezium/debezium/assets/379/c72fd88c-8dcd-4dff-81c8-8525a4f96108">